### PR TITLE
Refactor ModelRegion code 

### PR DESCRIPTION
**Pull request by cmcantalupo**
General clean up of the ModelRegion and ModelApplication code.  Should be no change to functionality.  Is now more testable, but no tests have been added.

### CODE REVIEW

File: src/NestedModelRegion.cpp [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+#include "config.h"
+#include <iostream>
+#include <mpi.h>
+
+#include "geopm.h"
+#include "geopm_time.h"
+#include "geopm_imbalancer.h"
+#include "Exception.hpp"
+#include "NestedModelRegion.hpp"
```
Comment (dianarg) [inline src/NestedModelRegion.cpp]: could move up

File: src/IgnoreModelRegion.cpp [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY LOG OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include <iostream>
+
+#include "geopm.h"
+#include "Exception.hpp"
+#include "IgnoreModelRegion.hpp"
```
Comment (dianarg) [inline src/IgnoreModelRegion.cpp]: could move up

File: src/DGEMMModelRegion.cpp [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY LOG OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include <iostream>
+#include <cmath>
+
+#include "geopm.h"
+#include "DGEMMModelRegion.hpp"
```
Comment (dianarg) [inline src/DGEMMModelRegion.cpp]: could move up

**Comment by cmcantalupo** [comment]
@dianarg I have addressed the review comments and Travis is also passing...Shall I squash the fixups and merge?

**Pull request merged**